libsoftwarecontainer/src/gateway/dbus/dbusgatewayinstance.cpp

```diff
@@ -55,12 +64,13 @@ DBusGatewayInstance::DBusGatewayInstance(ProxyType type,
     // Send an empty array as config for the irrelevant proxy type
     json_t* emptyArray = json_array();
     json_object_set(m_entireConfig, unusedTypeStr, emptyArray);
-
-
 }
 
 DBusGatewayInstance::~DBusGatewayInstance()
 {
+    // TODO: The "extra" teardown here is likely unnecessary. This should be removed
```

Comment:
No it's not nice. So I'd really like to "fix" this so the comment is moot but there's no time to do that thoroughly now. And as it was, i.e. uncommented, it was really difficult to understand the purpose and I had to waste time on finding people that knew about this. So I think keeping this info here adds some value. Not the best of solutions perhaps but at least it might save some time for the next person.